xxxx-ledger-plugin-interface-2/xxxx-ledger-plugin-interface-2.md

```diff
@@ -0,0 +1,502 @@
+
+###### Example
+
+``` js
+'2017-12-02T11:51:26.627Z'
+```
+
+#### Transfer#custom
+<code>**custom**:Object</code>
+
+Ledger plugins MAY use this object to accept and/or set additional fields for other features they support. The object MUST be serializable, i.e. only plain JSON types are allowed anywhere in the object or sub-objects.
+
+If the `custom` data is too large, the ledger plugin MUST reject with a `MaximumCustomDataSizeExceededError`.
+
+Note that connectors MAY forward some fields of `custom` data from plugin to plugin, but generally are not expected to. All `custom` fields that were passed to `sendTransfer` MUST be passed to the transfer handler by the plugin on the receiving side. The only exception are properties which start with the underscore character (`_`), which MAY be consumed by the plugin and not passed on.
+
+###### Example
+``` js
+{
+  // Starts with an underscore, consumed by plugin
+  _alternateAccount: 'bob-savings',
+  // Other property, passed on to next connector/receiver
+  executionPriority: 9
+}
+```
+
+## Class: FulfillmentInfo
+<code>class FulfillmentInfo</code>
+
+The `FulfillmentInfo` class is used to describe the fulfillment and associated data that is returned when a transfer successfully completes.
+
+###### Fields
+| Type | Name | Description |
+|:--|:--|:--|
+| `String` | [fulfillment](#fulfillmentinfofulfillment) | Cryptographic hold fulfillment |
+| `Buffer` | [data](#fulfillmentinfodata) | Transport-layer data |
+| `Object` | [custom](#fulfillmentinfocustom) | Object containing ledger plugin specific options |
+
+### Fields
+
+#### FulfillmentInfo#fulfillment
+<code>**fulfillment**:String</code>
+
+A cryptographic fulfillment that is the SHA-256 preimage of the hash provided as the [`executionCondition`](#transferexecutioncondition) when the transfer was first prepared.
+
+Fulfillments are base64url-encoded values with a length of exactly 32 bytes.
+
+Ledger plugins that do not support holds MUST reject with an `HoldsNotSupportedError` if this parameter is provided.
+
+#### FulfillmentInfo#data
+<code>**data**:Buffer</code>
+
+Transport-layer data that travels with the fulfillment.
+
+#### FulfillmentInfo#custom
+<code>**custom**:Object</code>
+
+Ledger plugins MAY use this object to accept and/or set additional fields for other features they support. The object MUST be serializable, i.e. only plain JSON types are allowed anywhere in the object or sub-objects.
+
+If the `custom` data is too large, the ledger plugin MUST reject with a `MaximumCustomDataSizeExceededError`.
+
+Note that connectors MAY forward some fields of `custom` data from plugin to plugin, but generally are not expected to. All `custom` fields that were passed to `sendTransfer` MUST be passed to the transfer handler by the plugin on the receiving side. The only exception are properties which start with the underscore character (`_`), which MAY be consumed by the plugin and not passed on.
```

Comment:
Replace: "All `custom` fields that were passed to `sendTransfer` MUST be passed to the transfer handler by the plugin on the receiving side"

With: "All `custom` fields that were set in `FulfillmentInfo#custom` on the receiving side MUST be set in `FulfillmentInfo#custom` when it is resolved by the promise on the sending side."